Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php

```diff
@@ -15,27 +17,57 @@ class ContentTypeForContentPanelStrategy extends AbstractNavigationPanelStrategy
     const ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT = 'ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT';
     const ROLE_ACCESS_DELETE_CONTENT_TYPE_FOR_CONTENT = 'ROLE_ACCESS_DELETE_CONTENT_TYPE_FOR_CONTENT';
 
-    /**
-     * @var ContentTypeRepositoryInterface
-     */
-    protected $contentTypeRepository;
-
-    /**
-     * @var ContextManager
-     */
-    protected $contextManager;
+    protected $contentTypes;
+    protected $datatableParameterNames;
 
     /**
      * @param ContentTypeRepositoryInterface $contentTypeRepository
      * @param ContextManager                 $contextManager
      * @param string                         $parent
      * @param int                            $weight
+     * @param array                          $defaultDataParameter
+     * @param TranslatorInterface            $translator
+     * @param TranslationChoiceManager       $translationChoiceManager
      */
-    public function __construct(ContentTypeRepositoryInterface $contentTypeRepository, ContextManager $contextManager, $parent, $weight)
+    public function __construct(
+        ContentTypeRepositoryInterface $contentTypeRepository,
+        ContextManager $contextManager,
+        $parent,
+        $weight,
+        $defaultDataParameter,
+        TranslatorInterface $translator,
+        TranslationChoiceManager $translationChoiceManager)
     {
-        parent::__construct('content_type_for_content', self::ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT, $weight, $parent);
-        $this->contentTypeRepository = $contentTypeRepository;
-        $this->contextManager = $contextManager;
+        $dataParameter = array();
+
+        $defaultDataParameter = $this->preFormatDatatableParameter($defaultDataParameter, $translator);
```

Comment:
i think not is a good idea to add this in constructor, you can move this code in a method